File: bixby/src/main/java/guepardoapps/lucahome/bixby/classes/actions/ApplicationAction.java
package guepardoapps.lucahome.bixby.classes.actions;

import android.support.annotation.NonNull;

import java.util.Locale;

@SuppressWarnings("WeakerAccess")
public class ApplicationAction implements IBixbyAction {
    private static final String Tag = ApplicationAction.class.getSimpleName();

    private String _packageName;

    public ApplicationAction(@NonNull String packageName) {
        _packageName = packageName;
    }

    public ApplicationAction() {
        this("");
    }

    public String GetPackageName() {
        return _packageName;
    }

    @Override
    public String GetDatabaseString() {
        return _packageName;
    }

    @Override
    public String GetInformationString() {
        return String.format(Locale.getDefault(), "%s \nfor %s", Tag, _packageName);
    }

    @Override
    public String toString() {
        return String.format(Locale.getDefault(),
                "{\"Class\":\"%s\",\"PackageName\":\"%s\"}",
                Tag, _packageName);
    }
}

File: bixby/src/main/java/guepardoapps/lucahome/bixby/models/actions/ApplicationAction.kt
package guepardoapps.lucahome.bixby.models.actions

import guepardoapps.lucahome.bixby.models.shared.IBixbyEntity

class ApplicationAction : IBixbyEntity {
    private val tag = ApplicationAction::class.java.simpleName

    lateinit var packageName: String

    override fun getDatabaseString(): String {
        return packageName
    }

    override fun getInformationString(): String {
        return "$tag: $packageName"
    }

    @Throws(NoSuchMethodException::class)
    override fun parseFromDb(databaseString: String) {
        throw NoSuchMethodException("This method is not available!")
    }

    override fun toString(): String {
        return "{" +
                "\"Class\":\"$tag\"," +
                "\"PackageName\":\"$packageName\"" +
                "}"
    }
}